<?xml version="1.0"?>
<launch>
	<!-- Some general parameters -->
	<rosparam file="$(find home)/param/ros.yaml"/>

    <node name="MapServer" pkg="map_server" type="map_server" output="screen" args="$(find home)/maps/home_map_20180717.yaml"/>
    <!-- <node name="MapServer" pkg="map_server" type="map_server" output="screen" args="$(find home)/maps/2ndFloor_map3.yaml"/> -->

    <!-- Start lidar and motor-->
    <include file="$(find turtlebot_bringup)/launch/minimal.launch" />
    <include file="$(find home)/launch/rplidar.launch" />
    <include file="$(find home)/launch/amcl.launch" />

    <!-- Start the Operator to control the simulated robot -->
    <node name="Operator" pkg="csc_nav2d_operator" type="operator" output="screen">
        <rosparam file="$(find home)/param/operator.yaml"/>
        <rosparam file="$(find home)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find home)/param/local_costmap_params.yaml" command="load" />
    </node>

	<!-- Start the Navigator to move the robot autonomously -->
	<node name="Navigator" pkg="csc_nav2d_navigator" type="navigator" >
        <rosparam file="$(find home)/param/navigator.yaml"/>
        <rosparam file="$(find home)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find home)/param/local_costmap_params.yaml" command="load" />
	</node>

    <node name="SetGoal" pkg="csc_nav2d_navigator" type="set_goal_client" />

	<!-- RVIZ to view the visualization -->
	<node launch-prefix="nice -10" name="RVIZ" pkg="rviz" type="rviz" args=" -d $(find home)/rviz/myCar.rviz" />
</launch>
